src/guiguts/maintext.py

```diff
@@ -435,6 +435,13 @@ def switch_text_peer(event: tk.Event) -> None:
         # Force focus to maintext widget
         self.after_idle(lambda: grab_focus(self.root, self, True))
 
+        # Initialize highlighting tags
+        from guiguts.highlight import (  # pylint: disable=import-outside-toplevel
```

Comment:
I would prefer we follow pylint's suggestion and only import at the top of the file. Is this actually revealing that we have circular imports between maintext and highlight? So far I have managed to avoid those, as I am fearful of a tangled web developing. Is there an alternative, or are we stuck with this?

Reply:
That's exactly it. So we can either import on demand in `Maintext.__init__` or an alternative is to move the `HighlightTag` and `HighlightColors` classes, and the `highlight_initialize_tags` method, into `maintext.py` or somewhere else that is already initialized, or this is needed to avoid a circular import exception.

This was also seen in #519 and I would assume also affects the imports in maintext.py for #525 and #528. (So basically all of the highlighting bits).

For **this** PR that would probably be a small number of things, but once you consider all the other highlighter work, I think we'd end up moving much or all of highlight.py to be part of maintext.py (because most things in highlight.py are calling `maintext()` for something or other). If you strongly want to avoid this type of import, I'm okay with doing that. One could argue that highlighting is just one of the attributes / activities of MainText.

Editorial note: in the prior sentence, when I was typing "attributes", Github or my browser or something suggested "atrocities" as an autocompletion 😆

Reply:
To be more direct, I will (in a different branch than this PR, for now) work on a refactor to move highlighting into maintext.py. It probably won't be that hard but we'll want to re-test *all* of the highlighters after.

Reply:
Handled in #544